internal/ir/xds.go

```diff
@@ -102,6 +102,8 @@ type HTTPListener struct {
 	TLS *TLSListenerConfig
 	// Routes associated with HTTP traffic to the service.
 	Routes []*HTTPRoute
+	// IsHTTP2 is set if the upstream client as well as the downstream server are configured to serve HTTP2 traffic.
+	IsHTTP2 bool
```

Comment:
@arkodg please take a moment to review the Kube API [conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) regarding bool fields.

Reply:
ha thanks for nudging me to think twice :), here's why I chose a `bool` for now
* most the HTTP logic in `gatewayapi` and `xds` can be reused between HTTP and GRPC APIs, reducing code complexity
* this is an internal API, and should be able to be changed easily w/o user intervention in case the `HTTPRoute` and `GRPCRoute` APIs completely diverge